qr-code/src/main/java/com/adyen/checkout/qrcode/DefaultQRCodeDelegate.kt

```diff
@@ -147,14 +149,18 @@ internal class DefaultQRCodeDelegate(
         }
 
         val viewType = when (action.paymentMethodType) {
-            PaymentMethodTypes.PAY_NOW -> QrCodeComponentViewType.FULL_QR_CODE
+            PaymentMethodTypes.PAY_NOW -> {
+                maxPollingDurationMillis = PAY_NOW_MAX_POLLING_DURATION
+                QrCodeComponentViewType.FULL_QR_CODE
+            }
             else -> QrCodeComponentViewType.SIMPLE_QR_CODE
```

Comment:
I think it would be nice to set the `maxPollingDurationMillis` back to the default value in the `else`